src/gpuhunt/providers/vultr.py

```diff
@@ -0,0 +1,205 @@
+import logging
+from typing import Any, Optional
+
+import requests
+from requests import Response
+
+from gpuhunt import QueryFilter, RawCatalogItem
+from gpuhunt._internal.constraints import KNOWN_AMD_GPUS, KNOWN_NVIDIA_GPUS
+from gpuhunt._internal.models import AcceleratorVendor
+from gpuhunt.providers import AbstractProvider
+
+logger = logging.getLogger(__name__)
+
+API_URL = "https://api.vultr.com/v2"
+
+EXCLUSION_LIST = ["GH200"]
+
+
+class VultrProvider(AbstractProvider):
+    NAME = "vultr"
+
+    def get(
+        self, query_filter: Optional[QueryFilter] = None, balance_resources: bool = True
+    ) -> list[RawCatalogItem]:
+        offers = fetch_offers()
+        return sorted(offers, key=lambda i: i.price) if offers is not None else []
+
+
+def fetch_offers() -> Optional[list[RawCatalogItem]]:
+    """Fetch plans with types:
+    1. Cloud GPU (vcg),
+    2. Bare Metal (vbm),
+    3. and others cpu plans, which includes:
+                             Cloud Compute (vc2),
+                             High Frequency Compute (vhf),
+                             High Performance (vhp),
+                             All optimized Cloud Types (voc)"""
+    try:
+        cloud_gpu_plans_response = _make_request("GET", "/plans?type=vcg")
+        bare_metal_plans_response = _make_request("GET", "/plans-metal")
+        other_plans_response = _make_request("GET", "/plans?type=all")
+        combined_response = {
+            "plans": (
+                cloud_gpu_plans_response.json().get("plans", [])
+                + other_plans_response.json().get("plans", [])
+            ),
+            "plans_metal": bare_metal_plans_response.json().get("plans_metal", []),
+        }
```

Comment:
(nit) Seems a bit redundant to build a dict just to immediately turn it into a list in `convert_response_to_raw_catalog_items`